lib/commands/context.js

```diff
@@ -362,6 +362,26 @@ extensions.onPageChange = async function onPageChange (pageChangeNotification) {
   let newIds = [];
   let newPages = [];
   let keyId = null;
+
+  // Checks if a URL is blacklisted in the 'safariIgnoreWebHostnames' capability
+  function isUrlBlacklisted (url, safariIgnoreWebHostnames) {
+    const blacklistedHosts = (safariIgnoreWebHostnames || '')
+      .split(',')
+      .map((b) => b.trim())
+      .filter((b) => !_.isEmpty(b));
+    for (const blacklistedHost of blacklistedHosts) {
+      if (blacklistedHost === url && url === 'about:blank') {
+        return true;
+      } else {
+        const hostname = new URL(url).hostname;
```

Comment:
would this throw an exception if the given value is not a valid url?

Reply:
Yes it would. I just tested it now. Good catch!